generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +208,123 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  @DataProvider(name = "with_typeref_field_in_record_pdsc")
+  private Object[][] createDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {"IsoDuration.pdsc", "PremiumService.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+            {"PremiumService.pdsc", "IsoDuration.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_typeref_field_in_record_pdsc")
+  public void testDataTemplateGenerationOrderWithResolver(String[] testArgs)
+      throws Exception
+  {
+    String[] pegasusFilenames = Arrays.copyOfRange(testArgs, 0, 2);
+    String[] javaFilenames = Arrays.copyOfRange(testArgs, 2, testArgs.length);
+    testDataTemplateGenerationDeterministic(pegasusFilenames, javaFilenames);
+  }
+
+  @DataProvider(name = "with_typeref_field_in_record_pdl")
+  private Object[][] createCasesWithTyperefFieldInRecord()
+  {
+    return new Object[][]
+        {
+            {"AField.pdl", "ARecord.pdl", "AField.java", "ARecord.java"},
+            {"ARecord.pdl", "AField.pdl", "AField.java", "ARecord.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_typeref_field_in_record_pdl")
+  public void testCasesWithTyperefFieldInRecord(String[] testArgs)
+      throws Exception
+  {
+    String[] pegasusFilenames = Arrays.copyOfRange(testArgs, 0, 2);
+    String[] javaFilenames = Arrays.copyOfRange(testArgs, 2, testArgs.length);
+    testDataTemplateGenerationDeterministic(pegasusFilenames, javaFilenames);
+  }
+
+  @DataProvider(name = "with_inline_field_in_record_pdl")
+  private Object[][] createCasesWithInlineFieldInRecord()
+  {
+    return new Object[][]
+        {
+            {"BRecord.pdl", "InlineRecord.java", "BRecord.java"},
```

Comment:
I think the only 2 cases you need are 

  {"BRecord.pdl", "InlineRecord.java"},
  {"InlineRecord.java", "BRecord.pdl"},"

Unless you want to test duplicates  like you have here.